### PR TITLE
init fixes

### PR DESCRIPTION
**Pull request by jesseditson**
Make --id required when programatically initializing apps, make --name work as expected both with and without --json

Fixes https://github.com/quip/quip-apps/issues/112
Fixes https://github.com/quip/quip-apps/issues/113

### CODE REVIEW

File: packages/quip-cli/src/commands/init.ts [hunk 1]
```diff
@@ -79,24 +85,32 @@ export default class Init extends Command {
         }),
         json: flags.boolean({
             char: "j",
-            description: "output responses in JSON (must provide --name)",
+            description:
+                "output responses in JSON (must provide --name and --id)",
```
Comment (gaspar09) [inline packages/quip-cli/src/commands/init.ts:89]: oclif provides a `dependsOn: ["name", "id"]` flag option for this

https://oclif.io/docs/flags

File: packages/quip-cli/src/commands/init.ts [hunk 2]
```diff
@@ -259,9 +275,9 @@ export default class Init extends Command {
         const dryRun = flags["dry-run"];
         const fetch = await cliAPI(flags.config, flags.site);
         const shouldCreate = !flags["no-create"];
-        if (flags.json && !flags.name) {
+        if (flags.json && (!flags.name || !flags.id)) {
             println(
-                chalk`{red You must provide --name when initializing with --json}`
+                chalk`{red You must provide --name and --id when initializing with --json}`
             );
             process.exit(1);
```
Comment (gaspar09) [inline packages/quip-cli/src/commands/init.ts:282]: see comment for `dependsOn`

**Review by gaspar09 — APPROVED** [review]

File: packages/quip-cli/src/commands/init.ts [hunk 3]
```diff
@@ -31,6 +31,8 @@ const defaultName = (dir?: string) => {
         .replace(/(:?^|\s)(\w)/g, (c) => c.toUpperCase());
 };
 
+const packageName = (name: string) => name.toLowerCase().replace(/\s+/g, "-");
```
[the remark this answers is not in the document]
Comment (jesseditson) [reply packages/quip-cli/src/commands/init.ts:34]: *trim ;) but yeah makes sense.

[the remark this answers is not in the document]
Comment (gaspar09) [reply packages/quip-cli/src/commands/init.ts:34]: I hate being bilingual

**Pull request merged**